lib/Cro/Tools/Template/ZeroMQWorkSinkService.pm6

```diff
@@ -41,47 +43,21 @@ class Cro::Tools::Template::ZeroMQWorkSinkService does Cro::Tools::Template {
             }
         }
         CODE
-
-        $file.spurt($entrypoint);
     }
 
-    sub write-cro-file($file, $id, $name, %options, @links) {
-        my $id-uc = env-name($id);
-        my $cro-file = Cro::Tools::CroFile.new(
-            :$id, :$name, :entrypoint<service.p6>, :entrypoints[
-                Cro::Tools::CroFile::Endpoint.new(
-                    id => 'zmq',
-                    name => 'ZeroMQ',
-                    protocol => 'tcp',
-                    host-env => $id-uc ~ '_HOST',
-                    port-env => $id-uc ~ '_PORT'
-                )
-            ], :@links
-        );
-        $file.spurt($cro-file.to-yaml());
+    method cro-file-endpoints($id-uc, %options) {
+        Cro::Tools::CroFile::Endpoint.new(
+            id => 'zmq',
+            name => 'ZeroMQ',
+            protocol => 'tcp',
+            host-env => $id-uc ~ '_HOST',
+            port-env => $id-uc ~ '_PORT'
+        ),
     }
 
-    sub write-meta($file, $name) {
-        my $m = META6.new(
-            name => $name,
-            description => 'Write me!',
-            version => Version.new('0.0.1'),
-            perl-version => Version.new('6.*'),
-            depends => <Cro::ZMQ>,
-            tags => (''),
-            authors => (''),
-            auth => 'Write me!',
-            source-url => 'Write me!',
-            support => META6::Support.new(
-                source => 'Write me!'
-            ),
-            provides => {},
-            license => 'Write me!'
-        );
-        spurt($file, $m.to-json);
-    }
+    method meta6-depends(%options) { <Cro::ZMQ> }
 
-    sub env-name($id) {
-        $id.uc.subst(/<-[A..Za..z_]>/, '_', :g)
-    }
+    method meta6-provides(%options) { () }
+
+    method meta6-resources(%options) { () }
```

Comment:
Given multiple implementations of these methods just return an empty list, perhaps the role should provide that as a default rather than forcing this implementation?